src/exo/pyparser.py

```diff
@@ -393,8 +393,6 @@ def parse_num_type(self, node, is_arg = False):
                     if isinstance(node.slice, pyast.Tuple):
                         dims = node.slice.elts
                     else:
-                        assert isinstance(node.slice,
```

Comment:
Why did you delete this assert?

Reply:
I believe I relaxed the constraint that dependent sizes are simple (i.e. only a name or constant). It seems we already have all the infrastructure to support expressions in tensor types but never removed this check. If you're concerned this might break something, maybe we can add some tests to increase our confidence?